orttraining/orttraining/eager/test/ort_ops.py

```diff
@@ -211,6 +211,38 @@ def test_log_softmax(self):
         assert torch.allclose(cpu_result_c, ort_result_c.cpu())
         assert torch.allclose(ort_result_a.cpu(), ort_result_c.cpu())
 
+    @parameterized.expand([param((2, 64, 256), 0), param((2, 64, 256), 1),
+                           param((2, 64, 256), -1), ((4096, 1024), 0),
+                           param((512, 8192), 1)])
+    def test_softmax_grad(self, input_shape, dim):
+        device = self.get_device()
+        cpu_tensor = torch.nn.Parameter(torch.rand(input_shape))
+        ort_tensor = torch.nn.Parameter(cpu_tensor.detach().clone().to(device))
+        cpu_result = torch.softmax(cpu_tensor, dim=dim)
+        ort_result = torch.softmax(ort_tensor, dim=dim)
+        cpu_loss = cpu_result.pow(2).sum()
+        ort_loss = ort_result.cpu().pow(2).sum()
```

Comment:
Isn't `ort_loss` on the cpu now because of the `.cpu()`? I think that means the `ort_loss.backward()` is on the cpu and not the ort.

Reply:
The pow() and sum() calculation are moved to cpu() and so only the backward for these pow() and sum() should run on cpu. The softmax backward should run on ort. I am able to hit the breakpoints in our [softmax backward implementation](https://github.com/microsoft/onnxruntime/blob/caabfcd9203ccf70e52aaeded315269efda61652/orttraining/orttraining/training_ops/cpu/op_gradients.cc#L71) with this code